modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/utils/MinMuleVersionResult.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.loader.parser.java.utils;
+
+import org.mule.runtime.api.meta.MuleVersion;
+
+/**
+ * Holds information about the minimum {@link MuleVersion} that a component can run on and the reason for that value.
+ *
+ * @since 4.6
+ */
+public class MinMuleVersionResult {
+
+  private String componentName;
+  private MuleVersion minMuleVersion;
+  private String reason;
+  private MinMuleVersionResult innerComponent = null;
+
+  public MinMuleVersionResult(String componentName, MuleVersion minMuleVersion, String reason) {
+    this.componentName = componentName;
+    this.minMuleVersion = minMuleVersion;
+    this.reason = reason;
+  }
+
+  public MuleVersion getMinMuleVersion() {
+    return minMuleVersion;
+  }
+
+  public String getReason() {
```

Comment:
Is the reason being logged somewhere? I haven't find it and don't know if maybe this is due to a discussion that already happened.

Reply:
You are right. It is not logged anywhere at the moment. We should discuss if we want to just log it in the ModelLoaderDelegates or do we extend the min mule version property and add the reason to it? What do you think?

Reply:
I don't think that information is relevant for the model, since it speak more of the implementation than the behavior or contract of the extension itself. If we choose to log it, yes, do it at model loader delegates level, so that all sdks get that logging.

Reply:
if not logged, why have it at all?